app/lib/backend/storage_rewrite.rb

```diff
@@ -63,7 +63,7 @@ def rewrite_provider(id)
       metrics = provider.metrics.includes(:service, :parent)
       usage_limits = provider.usage_limits.includes(plan: :service)
 
-      total_count = services.size + cinstances.size + metrics.size + usage_limits.size + bought_cinstances.size
+      total_count = services.size + cinstances.size + metrics.size + usage_limits.size + provider.bought_cinstances_count
```

Comment:
I wonder, `services.size`, `cinstances.size` etc ... they are loading the records but `bought_cinstances_count` is not loading `bought_cinstances`

Why this change? Don't we need `bought_cinstances` later on? If we do then I think there is no need to use the cached counter

Reply:
Also the counter cache is made so you do not have to use `#bought_cinstances_count` but plainly do `bought_cinstances.size`

> The method “size” will use the counter_cache if its present, where as using “.count” itself would do the actual sql count.